README.md

```diff
@@ -2,59 +2,179 @@
 
 ## Learning Goals
 
-- Explain the adapter pattern
-- Use the pattern in Java
+- Explain the adapter pattern.
+- Use the pattern in Java.
 
 ## Introduction
 
-Use the adapter pattern when you want to introduce a new interface between a
-client class and existing functionality because the existing functionality does
-not work as is for the client class but should not (or cannot) be modified to
-meet the new requirements.
+An **adapter** is a structural design pattern that acts as a connector between
+two incompatible interfaces. This pattern uses a single class to bridge the
+functionalities of two interfaces that otherwise could not be directly
+connected.
 
-Here are the highlights from the Adapter pattern:
+We use the adapter pattern when we want to introduce a new interface between a
+client class and existing functionality. This is done usually when the existing
+functionality does not work as is for the client class but should not (or cannot)
+be modified to meet the new requirements.
+
+Here are the highlights from the adapter pattern:
 
 - Lets a client class work with a subcomponent that it wouldn't otherwise be
-  able to work with because the interfaces are not compatible
-- Allows the original subcomponent to be used without modification
+  able to work with because the interfaces are not compatible.
+- Allows the original subcomponent to be used without modification.
 - Useful to avoid re-writing or modifying existing functionality but still
-  re-use it
+  re-use it.
+
+The adapter design pattern is appropriately named because it does just that -
+adapt the code through the use of a singular class so that everything (new and
+old) can function properly. To provide a real-world analogy, think of a power
+plug adapter that can be used when traveling to different countries. Different
+countries around the world have a different standard socket when you go to
+plug in your laptop, phone, or other electronic devices. To still use your
+power plugs, you can buy an adapter. This will allow you to use your original
+charger and have it plugged into a different country's socket.
 
 ## Implementation
 
-The `Camera` class we put together so far works for traditional SLR cameras, but
-it assumes a camera has film in it. What if we wanted to have a digital camera?
+One more time, let us consider our transportation example from the factory
```

Comment:
It might be helpful for the students to have a class diagram here. Can the factory pattern lesson diagram be reused here?